### ISSUE TITLE
[Bug]: After high contrast with white is activated, the close icon is invisible

### ISSUE THREAD

**Issue by hnjoshi** [carbon-design-system/carbon#10199]
### Package

carbon-components

### Browser

Chrome

### Package version

Latest version

## Components / Docs Affected

 
- ToastNotification

## Detailed Description
> Describe in detail the issue you're having.

Problem: After high contrast with white is activated, the close icon is invisible

<img width="897" alt="image" src="https://media.github.ibm.com/user/42629/files/e936d900-21f6-11ec-9e9d-d0a3ade1bf8f">

> What did you expect to happen? What happened instead? What would you like to see changed?

Close icon should look properly as below:
<img width="428" alt="image" src="https://media.github.ibm.com/user/42629/files/08ce0180-21f7-11ec-9f82-a7a5b13f4138">

> What offering/team/product do you work on? Any pressing ship or release dates we should be aware of?
dns-svcs

### CodeSandbox example

.

### Steps to reproduce

.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by hnjoshi** [comment]
Linked to https://github.ibm.com/ibmcloud/pal/issues/5280

**Comment by joshblack** [comment]
cc @dakahn when you get a sec, could you confirm that this is an issue in HCM?

**Comment by dakahn** [comment]
@hnjoshi can you talk a little more about what "high contrast with white" means? 

three big things we need to start triaging contrast issues are browser version, operating system, and tool used for high contrast (plugin vs native). 

Thanks

**Comment by hnjoshi** [comment]
![image](https://user-images.githubusercontent.com/8787607/148452319-434a449b-0609-4027-9da4-127d505935e4.png)
In Windows operating system if we choose "High contrast white" theme we can see the problem.
Chrome Version 96.0.4664.110 (Official Build) (x86_64)

**Comment by dakahn** [comment]
Confirmed this bug on Windows 11 using the white high contrast theme
![image](https://user-images.githubusercontent.com/40970507/148464094-04c2d715-5c6a-407a-9a24-bf406f6326c8.png)

Bug not present with black high contrast theme
![image](https://user-images.githubusercontent.com/40970507/148464165-3fec1326-f282-4276-ac22-f80204ad2cf3.png)

**Comment by dakahn** [comment]
doing a little further research it seems like our HCM styling is the culprit here. Disabling the svg fill and the X renders as expected
![image](https://user-images.githubusercontent.com/40970507/148464583-f5bdb491-bf00-4ea6-81ba-13965cf2a910.png)
![image](https://user-images.githubusercontent.com/40970507/148464597-d869ac5a-a713-413a-9bf4-a60eff65cd58.png)

**Comment by tw15egan** [comment]
@dakahn does removing that line have any effect on the dark HCM / normal styles?

**Comment by dakahn** [comment]
@tw15egan yeah I assume so. We'll need to get a little MORE specific for the various themes. Something like:
```
@media screen and (-ms-high-contrast: white-on-black) {
    ...
}
```

**Comment by abbeyhrt** [comment]
Looks like this issue is still occurring so reopening to investigate!

**Comment by wlinacdl** [comment]
Can I know when the issue could fix? Thanks.
